src/providers/cambridge/test_utils.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from utils import get_page_content, find_links_and_tags, download_file
+from bs4 import BeautifulSoup
+import os 
+
+@pytest.mark.vcr()
+def test_page_content_valid():
+    response = get_page_content('https://www.cambridge.org/core/services/librarians/kbart')
+    assert response.status_code == 200
+
+@pytest.mark.vcr()
+def test_page_content_invalid():
+    response = get_page_content('https://www.cambridge.org/core/services/librarians/kbartWontWork')
+    assert response.status_code == 404 
+
+@pytest.mark.vcr
+def test_found_links():
+    url = 'https://www.cambridge.org/core/services/librarians/kbart'
+    response = get_page_content(url)
+    soup = BeautifulSoup(response.text, 'html.parser')
+    found_links = find_links_and_tags(soup, ['computer science'], 'cambridge ebooks and partner presses: 2023 ')
+    assert not len(found_links) == 0
+
+@pytest.mark.vcr
+def test_download_file(tmp_path):
+    url = 'https://www.cambridge.org/core/services/aop-cambridge-core/kbart/create/bespoke/717854B1C18FD5D0B882344E83E6F52B'
+    desired_filename = 'computer science'
+    target_filepath = str(tmp_path) + '/'
```

Comment:
are you sure that you need + "/" ?  since I see you use os.path.join in line 30, it's not needed, because `os.path.join adds` "/" after every string you called function with.

With your solution `expected_filepath` has two backslashes after the value of `target_filepath`.